src/argilla_server/schemas/v1/questions.py

```diff
@@ -293,6 +246,73 @@ class SpanQuestionSettingsCreate(UniqueValuesCheckerMixin):
         min_items=SPAN_OPTIONS_MIN_ITEMS,
         max_items=SPAN_OPTIONS_MAX_ITEMS,
     )
+    visible_options: Optional[int] = Field(None, ge=SPAN_MIN_VISIBLE_OPTIONS)
+
+    @root_validator(skip_on_failure=True)
+    def check_visible_options_value(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        visible_options = values.get("visible_options")
+        if visible_options is not None:
+            num_options = len(values["options"])
+            if visible_options > num_options:
+                raise ValueError(
+                    "The value for 'visible_options' must be less or equal to the number of items in 'options'"
+                    f" ({num_options})"
+                )
+
+        return values
+
+
+class SpanQuestionSettingsUpdate(UpdateSchema):
+    type: Literal[QuestionType.span]
+    options: Optional[
+        conlist(
+            item_type=OptionSettings,
+            min_items=SPAN_OPTIONS_MIN_ITEMS,
+            max_items=SPAN_OPTIONS_MAX_ITEMS,
+        )
+    ]
+    visible_options: Optional[int] = Field(None, ge=SPAN_MIN_VISIBLE_OPTIONS)
```

Comment:
Why we are not running the same check that we have on create? (`check_visible_options_value`)

Reply:
Because options may not be informed. In that case, we cannot validate the value provided for `visible_options`. We need to validate the update value with what is already stored in the DB. This is related to my previous TODO line